@pacoorozco I got this for the first time after making these changes. Could it be caused by anything I did in the last commit?
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x70cd9d]

goroutine 12 [running]:
github.com/nmrshll/google-photos-api-client-go/lib-gphotos.(*Client).UploadFile(0xc0001d20e8, 0xc000560000, 0xa3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/nick/go/src/github.com/nmrshll/google-photos-api-client-go/lib-gphotos/client.go:122 +0x3ed
github.com/nmrshll/gphotos-uploader-cli/upload.(*FileUpload).upload(0xc0001d20c0, 0xc000405fa8, 0x1)
        /home/nick/go/src/github.com/nmrshll/gphotos-uploader-cli/upload/fileUpload.go:51 +0x9e
github.com/nmrshll/gphotos-uploader-cli/upload.StartFileUploadWorker.func1(0xc00014a048)
        /home/nick/go/src/github.com/nmrshll/gphotos-uploader-cli/upload/fileUpload.go:31 +0x75
created by github.com/nmrshll/gphotos-uploader-cli/upload.StartFileUploadWorker
        /home/nick/go/src/github.com/nmrshll/gphotos-uploader-cli/upload/fileUpload.go:29 +0x87
exit status 2
```

edit: yes it's this line 
```
			if batchResponse.ServerResponse.HTTPStatusCode == 429 {
```

--- a/lib-gphotos/client.go
+++ b/lib-gphotos/client.go
@@ -119,7 +119,8 @@ func (client *Client) UploadFile(filePath string, pAlbumID ...string) (*photosli
 		}).Do()
 		if err != nil {
 			// handle rate limit error by sleeping and retrying
-			if batchResponse.ServerResponse.HTTPStatusCode == 429 {
+			if batchResponse != nil && batchResponse.ServerResponse != nil &&
+				batchResponse.ServerResponse.HTTPStatusCode == 429 {
 				after, err := strconv.ParseInt(batchResponse.ServerResponse.Header.Get("Retry-After"), 10, 64)
 				if err != nil || after == 0 {
 					after = 10

--- a/lib-gphotos/client.go
+++ b/lib-gphotos/client.go
@@ -14,6 +14,7 @@ import (
 	"github.com/palantir/stacktrace"
 	"golang.org/x/oauth2"
 
+	"google.golang.org/api/googleapi"
 	photoslibrary "google.golang.org/api/photoslibrary/v1"
 )
 
@@ -119,9 +120,8 @@ func (client *Client) UploadFile(filePath string, pAlbumID ...string) (*photosli
 		}).Do()
 		if err != nil {
 			// handle rate limit error by sleeping and retrying
-			if batchResponse != nil && batchResponse.ServerResponse != nil &&
-				batchResponse.ServerResponse.HTTPStatusCode == 429 {
-				after, err := strconv.ParseInt(batchResponse.ServerResponse.Header.Get("Retry-After"), 10, 64)
+			if err.(*googleapi.Error).Code == 429 {
+				after, err := strconv.ParseInt(err.(*googleapi.Error).Header.Get("Retry-After"), 10, 64)
 				if err != nil || after == 0 {
 					after = 10
 				}

